container/kubernetes/onboarding/deviceonboarding.py

```diff
@@ -0,0 +1,396 @@
+#      without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+"""Gateway IOT Device OnBoarding Framework."""
+
+import sys
+import os.path
+import glob
+import argparse
+import subprocess
+import shutil
+import tempfile
+from collections import defaultdict
+from enum import Enum
+from git import Repo
+import yaml
+NAME_SPACE = None
+
+
+class CONFIG(Enum):
+    """ Enum class for deployment
+    and machine config separation.
+    """
+    DEPLOYMENT = 1
+    GATEWAY = 2
+
+
+class DEPLOYMENTTYPE(Enum):
+    """ Enum class for deployment
+    type differentiation.
+    """
+    CREATE = "create"
+    UPDATE = "update"
+    DELETE = "delete"
+
+
+def get_all_subdir_paths(dirpath):
+    """Return list of subdirectory paths for give directory."""
+    dirlist = []
+    for root, directories, _filename in os.walk(dirpath):
+        for dirs in directories:
+            dirlist.append(os.path.join(root, dirs))
+    return dirlist
+
+
+def check_deployment_status(deployment):
+    """Return application deployment status in a given k8s node."""
+    cmd = f"kubectl -n {NAME_SPACE} get deployment {deployment} | wc -l"
+    process = subprocess.Popen(
+        cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    check_deploy_status = process.communicate()[0].decode('utf-8').strip()
+
+    # Cosmetic extraction of data from subprocess command
+    if '\n' in check_deploy_status:
+        check_deploy_status = check_deploy_status.splitlines()[1]
+
+    deployment_status = False
+    if int(check_deploy_status) > 0:
+        deployment_status = True
+    return deployment_status
+
+
+def git_clone_tempdir(git_url, git_dir, git_tag,
+                      appname_dir, config_type):
+    """Method to checkout git repository based on git_tag
+       and copy to application deployment directory.
+    """
+    # Create temporary dir
+    tempdir = tempfile.mkdtemp()
+    # Clone into temporary dir
+    repo = Repo.clone_from(git_url, tempdir)
+    if git_tag:
+        repo.head.reset(git_tag, index=True, working_tree=True)
+    # Copy desired file from temporary dir
+    # if config_type DEPLOYMENT ENUM
```

Comment:
Recommend to delete this line, it is confusing.

Reply:
done.